### PR TITLE
telegram integration

### PR DESCRIPTION
**Pull request by Devorein**
Telegram integration PR like discord. Note that our prisma schema has been updated so its paramount to deploy the latest schema to prod. Similar to discord a `telegramUser` field is added to `User` and `TelegramUser` table is also created. The UI for connecting/disconnecting with telegram is the same as discord.

### CODE REVIEW

File: pages/api/profile/index.ts [hunk 1]
```diff
@@ -35,7 +35,8 @@ async function createUser (req: NextApiRequest, res: NextApiResponse<LoggedInUse
           }
         }
       },
-      discordUser: true
+      discordUser: true,
+      telegramUser: true
```
Comment (mattcasey) [inline pages/api/profile/index.ts:39]: make sure to remove this field when we write the user object to the session, like we do with discord

File: pages/api/telegram/connect.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+  const { id, ...rest } = telegramAccount;
+  const userId = req.session.user.id;
+  let telegramUser: TelegramUser;
+
+  try {
+    telegramUser = await prisma.telegramUser.create({
+      data: {
+        account: rest as any,
+        telegramId: String(id),
```
Comment (mattcasey) [inline pages/api/telegram/connect.ts:35]: curious why do you have to cast this as a string? i think it's ok to use a number if that's what telegram uses

File: charmClient.ts [hunk 3]
```diff
@@ -59,10 +60,8 @@ class CharmClient {
     });
   }
 
-  updateUser ({ addresses }: { addresses?: string[] }) {
-    return http.PUT<LoggedInUser>('/api/profile', {
-      addresses
-    });
+  updateUser (payload: { username?: string, avatar?: string, addresses?: string[] }) {
```
Comment (mattcasey) [inline charmClient.ts:63]: just want to confirm that we're not actually updating username/avatar from what i can tell, altho this change doesn't hurt anything :)

**Review by mattcasey — CHANGES_REQUESTED** [review]
one change request: don't write telegramuser data to session
one data model design suggestion: store telegram user id in its native type (number). this could get confusing if we are doing any other api calls, i think

File: components/common/PageLayout/components/Account/components/AccountModal/AccountModal.tsx [hunk 4]
```diff
@@ -19,10 +19,14 @@ import { useUser } from 'hooks/useUser';
 import styled from '@emotion/styled';
 import { CircularProgress, Tooltip } from '@mui/material';
 import charmClient from 'charmClient';
+// @ts-ignore
+import TelegramLoginButton from 'react-telegram-login';
```
Comment (mattcasey) [inline components/common/PageLayout/components/Account/components/AccountModal/AccountModal.tsx:23]: is it missing types? we should add a .d.ts file, i can help if you want

Comment (Devorein) [reply components/common/PageLayout/components/Account/components/AccountModal/AccountModal.tsx:23]: Wow I can do that. Never done that before so should be a learning experience. Good thing is that they have everything in proptypes so shouldn't take too long

File: pages/api/profile/index.ts (hunk 1 above)
Comment (Devorein) [reply pages/api/profile/index.ts:39]: Oh nice call. Thank you.

File: pages/api/telegram/connect.ts (hunk 2 above)
Comment (Devorein) [reply pages/api/telegram/connect.ts:35]: I was trying to play it safe as we don't have any idea about the range of this field. But I guess they are using a numeric id as well so its safe

**Comment by Devorein** [comment]
Both the changes have been implemented in the latest commits. I also made sure we don't allow the user to update their username and avatar using the API. I will be done via a first successful connection to discord/telegram only

**Review by mattcasey — APPROVED** [review]

**Pull request merged**